code/client/installhelper

```diff
@@ -0,0 +1,372 @@
+
+    # An instance of managedsoftwareupdate is running, so we need to try again later
+    while managedsoftwareupdate_pid:
+        log("*" * 60)
+        log("%s is running as pid %s." % (managedsoftwareupdate_pid,  os.getpid()))
+        log("checking again in %s seconds" % CHECK_TIME)
+        log("*" * 60)
+        time.sleep(CHECK_TIME)
+
+    # If managedsoftwareupdate is not running.. return from function to proceed
+    log("%s is not running, proceeding..." % (PROG_NAME))
+
+
+def reload_launch_daemons(launch_daemons):
+    '''Reloads Munki's Launch Daemons'''
+
+    # Log what we're upto
+    log("Reloading Launch Daemons")
+
+    # Retrieve all running launch daemons
+    cmd = ['/bin/launchctl', 'list']
+    cmd_out = subprocess.run(cmd, capture_output = True, check = False)
+    launchd_labels = split_launchctl_list(cmd_out)
+
+    # For each plist in /Library/LaunchDaemons/ prefixed with LAUNCHD_PREFIX
+    for launch_daemon in launch_daemons:
+        try:
+            # Make sure content is valid, before proceeding to reload
+            daemon_content = FoundationPlist.readPlist(launch_daemon)
+        except FoundationPlist.NSPropertyListSerializationException:
+            break
+        # Get the Label
+        daemon_label = daemon_content.get('Label')
+        # If we have a label
+        if daemon_label:
+            # Check to see if the launch daemon is loaded, before stopping
+            if daemon_label in launchd_labels:
+                # Stop the launch daemon at the login window
+                log("Stopping: %s" % launch_daemon)
+                cmd = ['/bin/launchctl', 'bootout', 'system/' + daemon_label]
+                subprocess.call(cmd)
+            # Load the launch daemon at the login window
+            log("Starting: %s" % launch_daemon)
+            cmd = ['/bin/launchctl', 'bootstrap', 'system', launch_daemon]
+            subprocess.call(cmd)
+
+
+def reload_loginwindow_launch_agents(launch_agents):
+    '''Reloads Munki's loginwindow Launch Agents'''
+
+    # Log what we're upto
+    log("Reloading LoginWindow Launch Agents")
+
+    # Retrieve all running launch agents at the login window
+    cmd = ['/bin/launchctl', 'list']
+    cmd_out = subprocess.run(cmd, capture_output = True, check = False)
+    launchd_labels = split_launchctl_list(cmd_out)
+
+    # For each plist in /Library/LaunchAgents/ prefixed with LAUNCHD_PREFIX
+    for launch_agent in launch_agents:
+        try:
+            # Make sure content is valid, before proceeding to reload
+            agent_content = FoundationPlist.readPlist(launch_agent)
+        except FoundationPlist.NSPropertyListSerializationException:
+            break
+        # If we're to limit loading
+        if agent_content.get('LimitLoadToSessionType'):
+            # If the launch agent includes LimitLoadToSessionType, and it contains LoginWindow
+            if 'LoginWindow' in agent_content.get('LimitLoadToSessionType'):
```

Comment:
These two if statements could be simplifed as one:
```python
if agent_content.get('LimitLoadToSessionType') == "LoginWindow":
```

Reply:
Simplified in - https://github.com/macmule/munki/commit/3a9302164f2ead8103e32ae94278de743c8dae22

But this was in place incase LimitLoadToSessionType contained multiple values (as it can).

Reply:
Oh, then, yes, that makes sense! It's an array of strings, and not a single string. OK.

Reply:
So my "simplification" is just plain wrong. Either go back to what you had, or do:
```python
if "LoginWindow" in agent_content.get('LimitLoadToSessionType', []):
```

Reply:
Reverted in: https://github.com/munki/munki/pull/1207/commits/8406c2bc5334fb898c4b412da16f3677f61c7001